### ISSUE TITLE
Request for documentation on the top-level appfwk CMakeLists.txt file

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/appfwk#50]
In trying to create the top-level CMakeLists.txt file for the example user package, I realized that it would be helpful to have a medium-level guide to what is in the standard appfwk CMakeLists.txt file. This could take the form of comments in the file, or a Wiki page that describes the different sections of the file. 

Some information that I would find useful:

-  what are the functions of the point_build_to() and MakeDataTypeLibraries() macros?

-  what are the scope(s) of the various set, add_xyz, and target_link_libraries calls (I'm guessing that their scope is the most-recent call to point_build_to(), but it would be nice to have that verified.)

In writing this, I realize that there are a number of ways to organize or view the the documentation that I'm thinking about. One is an explanation of the parts of the CMakeLists.txt file that is used in the appfwk repo. That might be useful as part of a tutorial on the appfwk and/or software coordination. A second type of documentation would be reference information on each of the CMake macros that are part of the DUNE DAQ environment, what they do, how to use them, etc. 
I recognize that the latter might be more appropriate for 'software coordination' documentation.  If someone wants me to create an Issue in a different repo, please let me know.

**Comment by jcfreeman2** [comment]
Documentation on appfwk's top-level CMakeLists.txt is available, along with other information, at https://github.com/DUNE-DAQ/appfwk/wiki/Creating-a-new-package

**Comment by bieryAtFnal** [comment]
John, Thanks for creating that Wiki page, it was very helpful.  Two comments...  

I added a line to the second bullet under "Your project's CMakeLists.txt file" to let the reader know that there will be more information coming about how to make use of DAQ_LIBRARIES_UNIVERSAL_EXE.

I stumbled over the paragraph that talks about MakeDataTypeLibraries.  Maybe that was because I didn't realize that the point of view had changed.  Instead of describing what is there, the text switches to giving suggestions about what the reader should do.  That, coupled with the fact that I don't yet know when or why I might be writing cmake code myself, left me wondering 'why do I need to know this now...?'
